tensor2tensor/data_generators/text_encoder.py

```diff
@@ -121,15 +121,25 @@ def vocab_size(self):
 class TokenTextEncoder(TextEncoder):
   """Encoder based on a user-supplied vocabulary."""
 
-  def __init__(self, vocab_filename, reverse=False, num_reserved_ids=2):
+  def __init__(self, vocab_filename, reverse=False, num_reserved_ids=2, unk=None):
```

Comment:
Please use our standard wordpiece-generator instead. It's now a commonly accepted method to get rid of the UNK problem, and it gives better results. On the other hand, if you want to generate vocab just from these files, feel free to change `_DATA_FILES` in `get_or_generate_vocab` or make it a parameter to the function.